docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,66 @@
+# Link Time Optimization
+
+Link Time Optimization (LTO) is a program memory usage optimization mechanism that the compiler performs at link time. At compile time, the compiler creates a special intermediate representation of all translation units. It then optimizes them as a single unit at link time, which uses less memory than non-LTO builds.
+
+## Using LTO in Mbed OS
+
+The Mbed OS build system implements LTO as an optional profile extension in `tools\profiles\extensions\lto.json`. Enabling LTO amends the build profile with LTO flags. LTO performs heavy memory optimizations that break debugging, so we don't recommend enabling it in debug or develop profiles.
```

Comment:
Would it make more sense to list the one profile we recommend it for, rather than the two we don't?

Reply:
updated